### PR TITLE
[#5124][#5146] feat(auth-ranger): Ranger plugin should support rename operation

### PR DESCRIPTION
**Pull request by xunliu**
### What changes were proposed in this pull request?

Add rename a securable object in the RoleChange.

### Why are the changes needed?

Fix: #5124 #5146

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

Add ITs.

### CODE REVIEW

**Comment by xunliu** [comment]
@jerqi Please help me review commit(https://github.com/apache/gravitino/pull/5176/commits/b1813c5dd88f183897e6cf4e5e03a1c106a0ae07) in this PR, Thanks.

**Comment by jerqi** [comment]
I am ok with the new interface.

File: api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+package org.apache.gravitino.authorization;
+
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.annotation.Evolving;
+
+/** The MetadataObjectChange interface defines the public API for managing roles in an authorization. */
+@Evolving
+public interface MetadataObjectChange {
```
Comment (jerqi) [inline api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java:28]: `RoleChange` and `MetadataObjectChange` is internal interface. Maybe we don't need to expose them in the api module.

File: authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java [hunk 2]
```diff
@@ -657,20 +698,386 @@ private boolean doRemoveSecurableObject(
...
+              /** COLUMN permission */
+              );
+      for (List<String> resNames : loop) {
+        removePolicyByMetadataObject(resNames);
+      }
+    }
+  }
+
+  /**
+   * Remove the TABLE, Need to rename these the relevant policies, `*.{table}`, `*.{table}.{column}`
```
Comment (jerqi) [inline authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java:772]: `rename` -> `delete`.

File: authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java [hunk 3]
```diff
@@ -657,20 +698,386 @@ private boolean doRemoveSecurableObject(
...
+        removePolicyByMetadataObject(rangerMetadataObject.names());
+        break;
+      default:
+        throw new IllegalArgumentException(
+            "Unsupported metadata object type: " + rangerMetadataObject.type());
+    }
+  }
+
+  /**
+   * Remove the SCHEMA, Need to rename these the relevant policies, `{schema}`, `{schema}.*`,
```
Comment (jerqi) [inline authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java:726]: `rename` -> `delete`?

**Review by jerqi — APPROVED** [review]
LGTM. some minors.

File: api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      this.newMetadataObject = newMetadataObject;
+    }
+
+    /**
+     * Returns the metadataObject to be renamed.
+     *
+     * @return return a metadataObject.
+     */
+    public MetadataObject getMetadataObject() {
+      return this.metadataObject;
```
Comment (FANNG1) [inline api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java:76]: please remove `this`

File: api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+      this.metadataObject = metadataObject;
+      this.newMetadataObject = newMetadataObject;
+    }
+
+    /**
+     * Returns the metadataObject to be renamed.
+     *
+     * @return return a metadataObject.
+     */
+    public MetadataObject getMetadataObject() {
```
Comment (FANNG1) [inline api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java:75]: remove `get` from method name?

File: authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java [hunk 6]
```diff
@@ -657,20 +698,314 @@ private boolean doRemoveSecurableObject(
     return true;
   }
 
+  /**
+   * IF remove the SCHEMA, Need to remove these the relevant policies, `{schema}`, `{schema}.*`,
+   * `{schema}.*.*` <br>
+   * IF remove the TABLE, Need to remove these the relevant policies, `{schema}.*`, `{schema}.*.*`
```
Comment (FANNG1) [inline authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java:704]: Need -> need

File: authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java (hunk 3 above)
Comment (xunliu) [reply authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java]: DONE.

File: authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java (hunk 2 above)
Comment (xunliu) [reply authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java]: DONE.

File: api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java (hunk 4 above)
Comment (xunliu) [reply api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java]: DONE.

File: api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java (hunk 5 above)
Comment (xunliu) [reply api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java]: DONE.

File: authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java (hunk 6 above)
Comment (xunliu) [reply authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java]: DONE.

File: api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    public MetadataObject metadataObject() {
+      return metadataObject;
+    }
+
+    /**
+     * Returns the new metadataObject object.
+     *
+     * @return return a metadataObject object.
+     */
+    public MetadataObject getNewMetadataObject() {
```
Comment (FANNG1) [inline api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java:84]: remove get too?

File: api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    private RemoveMetadataObject(MetadataObject metadataObject) {
+      this.metadataObject = metadataObject;
+    }
+
+    /**
+     * Returns the metadataObject to be renamed.
+     *
+     * @return return a metadataObject.
+     */
+    public MetadataObject getMetadataObject() {
```
Comment (FANNG1) [inline api/src/main/java/org/apache/gravitino/authorization/MetadataObjectChange.java:140]: get

**Comment by FANNG1** [comment]
LGTM except minor comments

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**